modules/upgrade-cluster-version-parameters.adoc

```diff
@@ -0,0 +1,179 @@
+the CVO uses the best update server for the cluster and region.
+
+|`ClusterVersionSpec.channel`
+|Optional. Specify the channel to use to apply non-standard updates to the
+cluster.
+
+[IMPORTANT]
+====
+The default channel contains stable updates. Do not specify a
+`ClusterVersionSpec.channel` value on production clusters.
+====
+
+|`ClusterVersionSpec.overrides`
+|Optional. A list of overides for components that the CVO manages. Mark
+components as `unmanaged` to prevent the CVO from creating or updating the object.
+[IMPORTANT]
+====
+Set the `ClusterVersionSpec.overrides` parameter value only during cluster
+debugging. Setting this value can prevent successful upgrades and is not
+suggested for production clusters.
+====
+
+|`ClusterVersionStatus`
+|The status of available updates and any in-progress updates. These values display
+the version that the cluster is reconciling to, and the conditions
+array reports whether the update succeeded, is in progress, or is failing.
+All of the `ClusterVersionStatus` values are set by the cluster itself, and you
+cannot modify them.
+
+|`ClusterVersionStatus.desired`
+|The version that the cluster is reconciling towards. In clusters that have not
+fully initialized, this value displays the available image or tag version data.
+
+|`ClusterVersionStatus.history`
+|A list of the most recent versions applied to the cluster. This value might be
+empty during cluster initialization and is updated when a new update is applied.
+The list is ordered by date, where the newest update is first in the list.
+Updates in the history have state `Completed` if the rollout completed and
+`Partial` if the update failed or did not complete.
+
+|`ClusterVersionStatus.observedGeneration`
+|The version of the spec that is synchronized. If this value is not equal to
+the `metadata.generation` value, then the `desired` and `conditions` values
+represent a previous cluster version.
+
+|`ClusterVersionStatus.versionHash`
+|A representation of the content that the cluster uses during upgrade.
+You cannot set this parameter value.
+
+|`ClusterVersionStatus.conditions`
+|The condition of the CVO. Conditions are valid for the current `desiredUpdate`
+only when the `metadata.generation` value matches the
+`status.generation` value.
+
+* `Available` means that the upgrade to the `desiredUpdate` value completed.
+* `Progressing` means that an upgrade is in progress.
+* `Failing` means that an update is blocked by a temporary or permanent error.
+
+The `ClusterVersionStatus.conditions` value displays both the reason that the
+cluster entered its current condition and a message that provides more
+information about the condition.
+
+|`ClusterVersionStatus.availableUpdates`
+|The list of appropriate updates for the cluster. This list is empty if no
+updates are recommended, the update service is unavailable, or you specified
+an invalid channel.
+
+|`UpdateState`
```

Comment:
They were in the [go code comments](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go#L122). @abhinavdahiya, where should I move this information?

Reply:
> From UpdateState to the last ClusterVersionList, I did not see them shown in clusterversion config

WDY mean by "I did not see them", are you talking about the clusterversion object in the cluster. The object you view in the cluster is serialized form of `ClusterVersion`. Fields that are missing could be effect of `omitempty`..

Reply:
> WDY mean by "I did not see them", are you talking about the clusterversion object in the cluster. The object you view in the cluster is serialized form of `ClusterVersion`. Fields that are missing could be effect of `omitempty`..

Yes, I'm talking about the clusterversion object in the cluster, which can be read by users directly. I thought it was about the clusterversion object. And take "UpdateState" for example, 
'''
Whether or not cluster updates succeeded.
CompletedUpdate means that an update succeeded.
PartialUpdate means that an update is in progress or did not finish.
'''
According to this description, it should be CompletedUpdate or PartialUpdate after a cluster update. It will mislead users to check whether upgrade succeed or not through "UpdateState" item, but they can not find it in clusterversion object. 
This was the reason why I raised them out. If this section "ClusterVersion parameters" is not about the clusterversion object, but the API used by CVO. Then @kalexand-rh @abhinavdahiya please can ignore all my comments about "ClusterVersion parameters".

Reply:
> The object you view in the cluster is serialized form of ClusterVersion. Fields that are missing could be effect of omitempty..
@abhinavdahiya ^^ Do u think it's better to add a note about what fields would be empty to avoid the misunderstand like "UpdateState"? Or should we give another section to describe the parameters in clusterversion object when introduce the upgrade process through CLI? Currently, QE as users much depends on the CV to check the upgrade status such as succeed or failed? failed for what reason? So this doc may be very important to help users on these.